### ISSUE TITLE
Need documentation on expected permissions for mounted volumes in a Pod

### ISSUE THREAD

**Issue by smanpathak** [kubernetes/kubernetes#73927]
**What happened**:
We ran percona mysql as Pod on GKE, Minikube and a proprietary PMK platform. The percona image is built with best practices docker documents. It runs the process mysqld as user "mysql" because running as root is not advised. It sets ownership of mysql data directory and declares it a "volume", which is an indication to underlying platform that it should not manage the data in it. What we are seeing is that this volume gets mounted as "root:root" on proprietary kubernetes and "mysql:root" on GKE/Minikube.
 
**What you expected to happen**:
All kubernetes platforms should exhibit the same behavior. I'm looking for standard documentation around how permissions are supposed to behave on kubernetes, the correct behavior to adhere to.

**How to reproduce it (as minimally and precisely as possible)**:
Run percona mysql docker image (percona/mysql-5.7) as a Pod

**Anything else we need to know?**:
The proprietary kubernetes uses datera storage as backend.
**Environment**:
- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.2", GitCommit:"17c77c7898218073f14c8d573582e8d2313dc740", GitTreeState:"clean", BuildDate:"2018-10-30T21:39:16Z", GoVersion:"go1.11.1", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.6", GitCommit:"9f8ebd171479bec0ada837d7ee641dec2f8c6dd1", GitTreeState:"clean", BuildDate:"2018-03-21T15:13:31Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by jayunit100** [comment]
Yeah, there can be some crippling differences w/ different volume types, thats for sure :) ... 

## Making it easier to troubleshoot vendor storage systems.

So, part of what is going on here is that the kubelet can be more explicit when mounting a volume about wether or not the ownership model is being modified and so on, I've started #73924  as an attempt at making the debugging process for this a little bit cleaner.
 
## Re: Consistent behaviour across filesystems
 
I'd be interested in knowing if anyone has discussed stratifying the common types of filesystems into categories.  IF so apps could easily rely on those categories in there volume definition and be confident that a storage class had that information in it.

**Comment by jayunit100** [comment]
After digging more into this ~ since fsGroup and supplmental Groups differ in their treatment depending on FS type, what we really need IMO are better logs at mounting time , hence this PR #73924 which gets us in that direction.

**Comment by smanpathak** [comment]
IMO, this behavior should be clearly defined and consistent regardless of storage provider types.

**Comment by jayunit100** [comment]
Agree, but we'd need to separate the semantics around Filesystems vs Block stores to do that, which would really make the user expeirience complex... Volume is a leaky abstraction, where all volumes aren't created equally, cross product with nodes and security policies, and you get a very complex matrix of possibilities . ...  wont always be able to honor fsGroup the same way,

good news is, the PR above will clarify exactly whats happening at runtime for logs that can be exchanged with storage vendors, which is the first step to making your life a little easier :)

**Comment by adeniyistephen** [comment]
@liggitt Hi, I would really love to work on this, please kindly provide input and pointers for the documentation. Thanks

**Comment by liggitt** [comment]
cc @msau42 for routing

**Comment by msau42** [comment]
@kubernetes/sig-storage-misc 
/assign @gnufied  

This is very related to all the fsgroup enhancements we are making and considering in the future.

**Comment by adeniyistephen** [comment]
> @kubernetes/sig-storage-misc
> /assign @gnufied
> 
> This is very related to all the fsgroup enhancements we are making and considering in the future.

OK, so it does not need documentation anymore? or you will be providing inputs sir?

**Comment by gnufied** [comment]
@adeniyistephen I would very much appreciate help in improving this area, but I am not sure what kind of input you are looking for. https://kubernetes.io/docs/tasks/configure-pod-container/security-context/ page documents it to some degree but it is not complete and needs expansion.

If you are familiar with this area then some of this is also documented in openshift docs https://docs.openshift.com/container-platform/3.11/install_config/persistent_storage/pod_security_context.html#fsgroup but we *do not* have SCC in k8s, so we can't use that stuff.

**Comment by adeniyistephen** [comment]
> @adeniyistephen I would very much appreciate help in improving this area, but I am not sure what kind of input you are looking for. https://kubernetes.io/docs/tasks/configure-pod-container/security-context/ page documents it to some degree but it is not complete and needs expansion.
> 
> If you are familiar with this area then some of this is also documented in openshift docs https://docs.openshift.com/container-platform/3.11/install_config/persistent_storage/pod_security_context.html#fsgroup but we _do not_ have SCC in k8s, so we can't use that stuff.

Yes i am familiar, but I just read the documentation, I think its fine for now, but like you said it needs expansion. so I was thinking if you have the input/pointers or explanation about what k8t would like to expand on it, then you can supply me so I do it for the team. thanks 😅
